plugins/XmlMaterialProfile/XmlMaterialProfile.py

```diff
@@ -369,8 +370,30 @@ def clearData(self):
         self._dirty = False
         self._path = ""
 
+    def getConfigurationTypeFromSerialized(self, serialized: str) -> Optional[str]:
+        return "material"
+
+    def getVersionFromSerialized(self, serialized: str) -> Optional[int]:
+        version = None
+        data = ET.fromstring(serialized)
+        metadata = data.iterfind("./um:metadata/*", self.__namespaces)
+        for entry in metadata:
+            tag_name = _tag_without_namespace(entry)
+            if tag_name == "version":
+                try:
+                    version = int(entry.text)
+                except Exception as e:
+                    raise InvalidInstanceError("Invalid version string '%s': %s" % (entry.text, e))
+                break
+        if version is None:
+            raise InvalidInstanceError("Missing version in metadata")
+        return version
+
     ##  Overridden from InstanceContainer
     def deserialize(self, serialized):
+        # update the serialized data first
+        from UM.Settings.Interfaces import ContainerInterface
```

Comment:
In general, import statements in non-module context are more expensive. This is one of the reasons tools like pylint and style checkers recommend against putting import statements after code.